eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -118,18 +121,49 @@ object NodeParams {
       .withFallback(ConfigFactory.parseFile(new File(datadir, "eclair.conf")))
       .withFallback(ConfigFactory.load())
 
-  def getSeed(datadir: File): ByteVector = {
-    val seedPath = new File(datadir, "seed.dat")
-    if (seedPath.exists()) {
-      ByteVector(Files.readAllBytes(seedPath.toPath))
-    } else {
-      datadir.mkdirs()
-      val seed = randomBytes32
-      Files.write(seedPath.toPath, seed.toArray)
-      seed
+  private def readSeedFromFile(seedPath: File): ByteVector = {
+    logger.info(s"use seed file: ${seedPath.getCanonicalPath}")
+    ByteVector(Files.readAllBytes(seedPath.toPath))
+  }
+
+  private def writeSeedToFile(path: File, seed: ByteVector): Unit = {
+    Files.write(path.toPath, seed.toArray)
+    logger.info(s"create new seed file: ${path.getCanonicalPath}")
+  }
+
+  private def migrateSeedFile(old: File, `new`: File): Unit = {
```

Comment:
nit: I'm not a fan of naming a variable `new` and having to use the backtick workaround...
Why not (to match `Files.copy`):

```suggestion
  private def migrateSeedFile(source: File, destination: File): Unit = {
```

Reply:
Done